src/Superpower/Parsers/Span.cs

```diff
@@ -258,5 +258,39 @@ public static TextParser<TextSpan> MatchedBy<T>(TextParser<T> parser)
                     result.Remainder);
             };
         }
+
+        /// <summary>
+        /// Parse as much of the input as matches <paramref name="text" />.
+        /// </summary>
+        /// <param name="text">The text to match until. The text is not included in the result.</param>
+        /// <returns>A parser that will match everything until the text argument is matched.</returns>
+        /// <exception cref="ArgumentNullException">The text is null.</exception>
+        public static TextParser<TextSpan> Until(string text)
+        {
+            if (text == null) throw new ArgumentNullException(nameof(text));
+
+            var expectation = $"Until expected '{text}'";
+
+            return (TextSpan input) =>
+            {
+                TextSpan remainder = input;
+                var matchIndex = 0;
+
+                while (remainder.Length>0)
```

Comment:
Still planning to use `IndexOf()` here?

Reply:
By IndexOf(), you mean the string method right?
If so, I was wondering if allocating new strings every time we visit, would be a better choice than just using the TextSpan consume/reminder pattern. As a general thought and related to my last question, I think I would have chosen a design with a mutable TextSpan for performance reasons. I'm not sure that I understand the need for immutability without also knowing a lot more of the details behind your design choices - but it's just what comes to mind when you start thinking about it as an outsider. If you create a new TextSpan you could still leave it untouched without advancing it (for use as a marker), so it still puzzles me a little that it has to be immutable. I would love to hear why, but don't write a long story - I'm just curious :-)